mkosi/machine.py

```diff
@@ -122,9 +143,55 @@ def run(self, commands: Sequence[str], timeout: int = 900, check: bool = True) -
 
         return process
 
-    def __exit__(self, *args: Any, **kwargs: Any) -> None:
+    def _kill(self, *args: Any, **kwargs: Any) -> None:
         if self._serial:
             self._serial.kill(signal.SIGTERM)
             self.exit_code = self._serial.wait()
             self._serial = None
         self.stack.__exit__(*args, **kwargs)
+
+    def __exit__(self, *args: Any, **kwargs: Any) -> None:
+        self._kill(*args, **kwargs)
+
+    def reboot(self) -> None:
+        self._kill(None, None, None)
+        self._boot()
```

Comment:
Given the comment below that reboot() doesn't really fit for unittest since we end up unnecessarily booting the machine, let's remove the reboot() method here as well. While it works, I think we should leave the opportunity open to add a reboot() method in the future that does not throw away all the state of the image (in other words, it would keep the temporary copy of the image intact when --ephemeral is used and also keeps the copy of efivars intact when qemu is used, which we don't do today). Note that you don't need to implement this reboot() method now, we can add the new version of it when we need it.